### DISCUSSION TITLE
Changing point of view of a fisheye image

### DISCUSSION THREAD

**Discussion by jumpjack** [ImageMagick/ImageMagick#5989] [Digital Image Processing] [answered]
I have a fisheye camera with FOV=180 pointing to a drawing on a flat table.

Question 1: Can I use IM to transform the resulting fisheye image into the flat drawing which is on the surface of the table?

Question 2: Can I transform the fisheye image to simulate the camera being moved at a different distance from the table?

**Comment by snibgo** [comment, marked as answer]
If your camera is an ideal equidistant fisheye, an arctan function will transform the image to rectilinear. See [De-barrel distortion: de-fisheye](http://im.snibgo.com/debarrel.htm#defish). If your camera isn't an ideal fisheye, you need to know (or estimate) the transfer function. I think that answers your Question 1.

For Question 2: convert the image to rectilinear, reduce the size, and convert back to fisheye.

**Comment by fmw42** [comment]
You can do fisheye processing with ffmpeg. It has a good module for that.

**Reply by jumpjack** [reply]
I could not find the right procedure to simulate camera movement starting from a fisheye image, using ffmpeg.

**Comment by fmw42** [comment]
See https://ffmpeg.org/ffmpeg-filters.html at the section 11.265 v360. Perhaps the h_offset or v_offset might do what you want.

**Comment by jumpjack** [comment]
I had another idea: I found the source image used to generate the fisheye image I was using:

[HP-SSA-DISR-2-3-EDR-RDR-V1.3\EXTRAS\MOSAICS\MOSIACS_PNG](https://archives.esac.esa.int/psa/ftp/CASSINI-HUYGENS/DISR/HP-SSA-DISR-2-3-EDR-RDR-V1.3/EXTRAS/MOSAICS/MOSIACS_PNG/)

I would like to create video like this from those images:

https://sci.esa.int/web/cassini-huygens/-/39218-huygens-descent-to-titan-surface

But I have just 12 frames, and I need to make the video much smooter; virtualdub interpolate filter does not help; can ImageMagick or FFMPEG create the "missing frames"? It's just a matter of zooming an image with upper number, but maintaining the same size, and keeping zooming until the covered area of, for example, 2.png, becomes the same of 1.png.

Then I have just to figure out which is the projection used for these mosaics... Stereographic? Equirectangular? Mercator?

**Reply by fmw42** [reply]
Looks to me like it is a 3D perspective like projection onto digital terrain data (image).  Do you have the terrain image?

Imagemagick has a morph module, but I do not know if that will do what you want.  FFMPEG should be able to do that also.

**Reply by jumpjack** [reply]
I have only those images.
